pkg/linter/lintRuleSyntacticRedundantCommon.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package linter
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/np-guard/models/pkg/connection"
+	"github.com/np-guard/models/pkg/ipblock"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
+)
+
+const NetworkACL = "network ACL" // todo: https://github.com/np-guard/vpc-network-config-analyzer/issues/724
+
+// a rule that is implied syntactically redundant:
+// nacl rule - implied by higher priority rules in the nacl table
+// sg rule - implied by other rules in the sg table
+type ruleRedundant struct {
+	rule         vpcmodel.RuleOfFilter
+	containRules map[int]*vpcmodel.RuleOfFilter // indexes of rules because of which this rule is redundant to their description
```

Comment:
containRules better be a set.

Reply:
The map is so that we can easily check whether a rule is already there and avoid adding it again.
I don't see a smoother way of doing this. (we can take in via webex)